pkg/controller/controller_utils.go

```diff
@@ -525,6 +525,43 @@ func (f *FakePodControl) Clear() {
 	f.Templates = []api.PodTemplateSpec{}
 }
 
+// ByLogging allows custom sorting of pods so the best one can be picked for getting its logs.
+type ByLogging []*api.Pod
+
+func (s ByLogging) Len() int      { return len(s) }
+func (s ByLogging) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+
+func (s ByLogging) Less(i, j int) bool {
+	// 1. assigned < unassigned
+	if s[i].Spec.NodeName != s[j].Spec.NodeName && (len(s[i].Spec.NodeName) == 0 || len(s[j].Spec.NodeName) == 0) {
+		return len(s[i].Spec.NodeName) > 0
+	}
+	// 2. PodRunning < PodUnknown < PodPending
+	m := map[api.PodPhase]int{api.PodPending: 0, api.PodUnknown: 1, api.PodRunning: 2}
+	if m[s[i].Status.Phase] != m[s[j].Status.Phase] {
+		return m[s[i].Status.Phase] > m[s[j].Status.Phase]
+	}
+	// 3. ready < not ready
```

Comment:
In the pod level this is "previous container"
In the deployment level, this is "previous deployment" (or at least that's what we have now in Origin)

I am not sure it would be clear to have -p do what you want. Once pods are scheduled, I guess we can have unready pods be prioritized over ready.